packages/amplify-graphql-relational-transformer/src/graphql-belongs-to-transformer.ts

```diff
@@ -158,6 +159,14 @@ export class BelongsToTransformer extends TransformerPluginBase {
 
 const validate = (config: BelongsToDirectiveConfiguration, ctx: TransformerContextProvider): void => {
   const { field, object } = config;
+  if (!ctx.transformParameters.allowGen1Patterns) {
+    if (field.type.kind === Kind.NON_NULL_TYPE) {
+      throw new InvalidDirectiveError(`@${BelongsToDirective.name} cannot be used on required fields.`);
```

Comment:
nit: Pointing customer to the faulty field by including its name along with parent type name can help them debug faster.